parse_test.go

```diff
@@ -743,6 +756,16 @@ func TestParse(t *testing.T) {
 				if !got.Equal(tc.t) {
 					t.Errorf("expected: %v, got: %v", tc.t, got)
 				}
+				if tc.sameZone {
```

Comment:
We'd better always check the zone equality. Only one existing test case (`2020-07-24 23:14:15 +0000`) should be fixed and then all cases should pass.